os/net/app-layer/coap/coap.c

```diff
@@ -497,57 +502,34 @@ coap_parse_message(coap_message_t *coap_pkt, uint8_t *data, uint16_t data_len)
     option_delta = current_option[0] >> 4;
     option_length = current_option[0] & 0x0F;
     ++current_option;
-    if(current_option >= data + data_len) {
-      /* Malformed CoAP - out of bounds */
-      LOG_WARN("BAD REQUEST: option delta outside message buffer\n");
-      return BAD_REQUEST_4_00;
-    }
 
     if(option_delta == 13) {
+      CHECK_OPTION_BOUNDARY(1);
       option_delta += current_option[0];
       ++current_option;
     } else if(option_delta == 14) {
       option_delta += 255;
+      CHECK_OPTION_BOUNDARY(2);
```

Comment:
Isn't it better to consistently check the boundary as soon as it is clear that will be required? Will make it easier to review since boundary check comes first, and also happens to save a little bit of CPU when the check fails.